pkg/api/v1/project_teams.go

```diff
@@ -27,14 +27,17 @@ type Team struct {
 	Roles []TeamRole `json:"roles"`
 }
 
-func (in *Team) ToAtlas(teamID string) *mongodbatlas.ProjectTeam {
-	result := &mongodbatlas.ProjectTeam{
-		TeamID:    teamID,
-		RoleNames: make([]string, 0, len(in.Roles)),
+func (in *Team) ToAtlas(teamID string) admin.TeamRole {
+	roleNames := make([]string, 0, len(in.Roles))
+	result := admin.TeamRole{
+		TeamId:    &teamID,
+		RoleNames: &roleNames,
 	}
 
 	for _, role := range in.Roles {
-		result.RoleNames = append(result.RoleNames, string(role))
+		roles := result.GetRoleNames()
+		roles = append(roles, string(role))
+		result.SetRoleNames(roles)
 	}
```

Comment:
Wouldn't that be easier?
```
	for _, role := range in.Roles {
		roleNames = append(roleNames, string(role))
	}
	result.SetRoleNames(roleNames)
```
In this way, we won't call Get and Set on `result` on every loop iteration.